pkg/provider/gke/gke.go

```diff
@@ -426,6 +426,33 @@ func (c *GKE) nodePoolRunning(zone, projectID, clusterID, poolName string) (bool
 	return false, nil
 }
 
+// NodePoolCheck returns an error if any of the nodepool do not exist.
+func (c *GKE) NodePoolCheck(*kingpin.ParseContext) error {
+	reqC := &containerpb.CreateClusterRequest{}
+
+	exists := true
+
+	for _, deployment := range c.gkeResources {
+		if err := yamlGo.UnmarshalStrict(deployment.Content, reqC); err != nil {
+			log.Fatalf("error parsing the cluster deployment file %s:%v", deployment.FileName, err)
+		}
+
+		for _, node := range reqC.Cluster.NodePools {
+			isRunning, err := c.nodePoolRunning(reqC.Zone, reqC.ProjectId, reqC.Cluster.Name, node.Name)
+			if err != nil {
+				log.Fatalln("error fetching nodePool info")
```

Comment:
I think we should start moving away form this aproach and just return the errors. Can't remember why I used `log.Fatal` in the other places.
```suggestion
				return errors.New("error fetching nodePool info")
```

If it is possible to have it like this please update the other places in this PR and will refactor the globally in another PR.